vita3k/renderer/include/renderer/vulkan/state.h

```diff
@@ -85,8 +85,8 @@ struct VKState : public renderer::State {
 
     VKState(int gpu_idx);
 
-    bool init(const char *base_path, const bool hashless_texture_cache) override;
-    bool create(SDL_Window *window, std::unique_ptr<renderer::State> &state, const char *base_path, const Config &config);
+    bool init(const char *shared_path, const bool hashless_texture_cache) override;
```

Comment:
The path used in init is for the builtin shaders. Builtin shaders are read-only files bundled along the Vita3K executable. As such base_path should be used and not shared_path. (this needs to be modified along all the calls leading to the builtin shader being created also).

Reply:
shared_path and base_path point to the same directory under Windows, but under Linux the two paths are different.

shared_path, as I had originally wrote it, points to what would effectively be /usr/share/Vita3K under Linux. I.e. a read only directory for _shared_ static resource files.

Under Windows this would be something like C:\Program Files\Vita3K (or where ever the user extracted the download to), but under Linux /usr/share is not a valid path for main executable binaries. As it violates the [FHS](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard). Though some scripts and subprograms can wind up here depending on the distro, the main binary invoked by the end-user is typically in /usr/bin. Which cannot have resource files stored there, and as such required a separate path to be used for compatibility.

Reply:
i have never see user extyract archive inside program file lol
is more deskop or in other drive ^^

Reply:
> shared_path and base_path point to the same directory under Windows, but under Linux the two paths are different.
> 
> shared_path, as I had originally wrote it, points to what would effectively be /usr/share/Vita3K under Linux. I.e. a read only directory for _shared_ static resource files.
> 
> Under Windows this would be something like C:\Program Files\Vita3K (or where ever the user extracted the download to), but under Linux /usr/share is not a valid path for main executable binaries. As it violates the [FHS](https://en.wikipedia.org/wiki/Filesystem_Hierarchy_Standard). Though some scripts and subprograms can wind up here depending on the distro, the main binary invoked by the end-user is typically in /usr/bin. Which cannot have resource files stored there, and as such required a separate path to be used for compatibility.

Thank you for the detailed explanation. Indeed that kind of is an issue. However, many linux user (on ubuntu or other distros) still use Vita3K by downloading the zip from the website, decompressing it and launching the ./Vita3K file inside it. Setting the data folder to be in the shared folder would break it in this case. I think a way to solve it would be to check if the data folder (or some of the builtin shader) is in the base path, in this case use the base path for all of this. Otherwise use the shared path for it. Or do you have another way of doing this? As to my knowledge the shared folder is not populated by default.

Reply:
>As to my knowledge the shared folder is not populated by default.

As an example, (although I need to rebase it), the [AppImage build](https://github.com/redpolline/Vita3K/tree/Impl_appimage) I made uses {squashfs root}/usr/share/Vita3K for it's shared_path. In it's [apprun.sh](https://github.com/redpolline/Vita3K/blob/591b070eabba845263e417b83da1591f8c3fdf89/appimage/apprun.sh#L1) it sets XDG_DATA_DIRS when invoking Vita3K so that it can find the mounted squashfs's shared_path.

Another packaging format which uses a filesystem container would probably get it's data from /usr/share/Vita3K as the default PREFIX used by pretty much any build system for production is /usr, and the maintainer would likely place the shared resources there by default in accordance with their policies.

Regardless, I've implemented checking the base_dir for the requisite shared folders and using it by default if they are found. (This will allow portable apps to work. While also allowing the end user or a script to change the default via XDG_DATA_DIRS if desired.)

One thing that could be useful though is implementing some kind of revision check for the shared data. So if a newer version of the shared data is shipped, but the older data still exists from some older installation, the correct path will be used. (As it is currently, if the data doesn't load properly, Vita3K just crashes.)